src/main/java/com/cisco/trex/stateful/api/lowlevel/ASTFTemplate.java

```diff
@@ -60,6 +60,11 @@ public ASTFTemplate(ASTFTCPClientTemplate astfTcpClientTemplate, ASTFTCPServerTe
         if (astfTcpClientTemplate.isStream() != astfTcpServerTemplate.isStream()) {
             throw new IllegalStateException(String.format(" Client template stream mode is %s and different from server template mode %s", astfTcpClientTemplate.isStream(), astfTcpServerTemplate.isStream()));
         }
+
+        if(tgName != null && (tgName.length() > 20 || tgName.isEmpty())){
+            throw new IllegalArgumentException(String.format("tgName %s is empty or too long", tgName));
+        }
```

Comment:
What's the original logic in python implementation? can you put it here to be compared?

Reply:
here is the code snippet in python file:
def __init__(self, client_template=None, server_template=None, tg_name=None):
        """
        Define a ASTF profile

        You should give either templates or cap_list (mutual exclusion).

        :parameters:
                  client_template  : ASTFTCPClientTemplate see :class:`trex.astf.trex_astf_profile.ASTFTCPClientTemplate`
                       client side template info

                  server_template  :  ASTFTCPServerTemplate see :class:`trex.astf.trex_astf_profile.ASTFTCPServerTemplate`
                       server side template info

                  tg_name          : string or None
                        All the templates within the tg_name (template group name) will have shared statistics. 
                        All the templates that haven't defined tg_name are collected under the same group.
        """
        ver_args = {"types":
                    [{"name": "client_template", 'arg': client_template, "t": ASTFTCPClientTemplate},
                     {"name": "server_template", 'arg': server_template, "t": ASTFTCPServerTemplate},
                     {"name": "tg_name", 'arg': tg_name, "t": str, "must": False}]
                    }
        ArgVerify.verify(self.__class__.__name__, ver_args)

        if client_template.is_stream() != server_template.is_stream() :
            raise ASTFError(" Client template stream mode is {0} and different from server template mode {1}".format( client_template.is_stream(), server_template.is_stream() ) )

        if tg_name is not None:
            if len(tg_name) > 20 or len(tg_name) == 0:
                raise ASTFError("tg_name is empty or too long")

        self.tg_name = tg_name
        self.fields = {}
        self.fields['client_template'] = client_template
        self.fields['server_template'] = server_template